File: src/main/java/org/activecraft/activecraftcore/manager/NickManager.java
package org.activecraft.activecraftcore.manager;

import org.activecraft.activecraftcore.events.ColornickEvent;
import org.activecraft.activecraftcore.events.NickEvent;
import org.activecraft.activecraftcore.playermanagement.Profilev2;
import org.activecraft.activecraftcore.events.NickEvent;
import org.activecraft.activecraftcore.playermanagement.Profilev2;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;

public class NickManager {

    public static void nick(Profilev2 profile, String nickname) {
        //call event
        NickEvent event = new NickEvent(profile, nickname);
        Bukkit.getPluginManager().callEvent(event);
        if (event.isCancelled()) return;

        profile.setRawNickname(nickname);
        profile.getDisplayManager().updateDisplayname();
    }

    public static void colornick(Profilev2 profile, ChatColor color) {
        //call event
        ColornickEvent event = new ColornickEvent(profile, color);
        Bukkit.getPluginManager().callEvent(event);
        if (event.isCancelled()) return;

        profile.setColorNick(color);
        profile.getDisplayManager().updateDisplayname();
    }

}

File: src/main/java/org/activecraft/activecraftcore/manager/NickManager.kt
package org.activecraft.activecraftcore.manager

import org.activecraft.activecraftcore.events.ColornickEvent
import org.activecraft.activecraftcore.events.NickEvent
import org.activecraft.activecraftcore.playermanagement.Profile
import org.bukkit.Bukkit
import org.bukkit.ChatColor

object NickManager {
    fun nick(profile: Profile, nickname: String) {
        val event = NickEvent(profile, nickname)
        Bukkit.getPluginManager().callEvent(event)
        if (event.cancelled) return
        profile.rawNickname = nickname
        profile.displayManager.updateDisplayname()
    }

    fun colornick(profile: Profile, color: ChatColor) {
        val event = ColornickEvent(profile, color)
        Bukkit.getPluginManager().callEvent(event)
        if (event.cancelled) return
        profile.colorNick = color
        profile.displayManager.updateDisplayname()
    }
}